src/corvus_python/pyspark/utilities/spark_utils/local_spark_utils.py

```diff
@@ -62,6 +64,39 @@ def getSecretWithLS(self, linked_service: str, secret_name: str) -> str:
                 raise ValueError(
                     f"Unknown secret type {target_secret.get('type')}")
 
+    def getToken(self, audience: str) -> str:
+        scopes = {
+            "Storage": "https://storage.azure.com/.default",
+            "Vault": "https://vault.azure.net/.default",
+            "AzureManagement": "https://management.azure.com/.default",
+            "DW": "https://database.windows.net/.default",
+            "Synapse": "https://dev.azuresynapse.net/.default",
+            "DataLakeStore": "https://datalake.azure.net/.default",
+            "DF": "https://datafactory.azure.net/.default",
+            "AzureDataExplorer": "https://kusto.kusto.windows.net/.default",
+            "AzureOSSDB": "https://ossrdbms-aad.database.windows.net/.default",
+        }
+
+        scope = scopes.get(audience)
+
+        if not scope:
+            raise ValueError(f"Unsupported audience '{audience}'")
+
+        tenant_id = None
+        get_token_config = self.config.get("getToken")
+        if get_token_config:
+            tenant_id = get_token_config.get("tenantId")
+
+        credential = AzureCliCredential()
+
+        try:
+            token = credential.get_token(scope, tenant_id=tenant_id)
```

Comment:
This still works if the `tenant_id` is `None`?

Reply:
Yup